src/main/java/ee/tuleva/onboarding/conversion/UserConversionService.java

```diff
@@ -171,44 +173,48 @@ private boolean isTransfersComplete(
+        .flatMap(Collection::stream)
         .filter(
-            transferExchange ->
-                transferExchange.getTargetFund() != null
-                    && transferExchange
-                        .getTargetFund()
-                        .getFundManager()
-                        .getName()
-                        .equalsIgnoreCase(CONVERTED_FUND_MANAGER_NAME)
-                    && amountMatches(transferExchange, fundBalances))
-        .map(transferExchange -> transferExchange.getSourceFund().getIsin())
+            exchange -> isConvertedFundManager(exchange) && amountMatches(exchange, fundBalances))
+        .map(exchange -> exchange.getSourceFund().getIsin())
         .collect(toList());
   }
 
-  private boolean amountMatches(TransferExchange transferExchange, List<FundBalance> fundBalances) {
-    if (transferExchange.getPillar() == 2) {
-      return transferExchange.getAmount().intValue() == 1;
+  private boolean isConvertedFundManager(Exchange exchange) {
+    return exchange
+        .getTargetFund()
+        .getFundManager()
+        .getName()
+        .equalsIgnoreCase(CONVERTED_FUND_MANAGER_NAME);
+  }
+
+  private boolean amountMatches(Exchange exchange, List<FundBalance> fundBalances) {
+    if (exchange.getPillar() == 2) {
+      return exchange.getAmount().intValue() == 1; // 100%
     }
-    if (transferExchange.getPillar() == 3) {
-      FundBalance fundBalance = fundBalance(transferExchange, fundBalances);
-      return transferExchange.getAmount().equals(fundBalance.getTotalValue());
+    if (exchange.getPillar() == 3) {
+      FundBalance fundBalance = fundBalance(exchange, fundBalances);
+      return exchange.getAmount().equals(fundBalance.getTotalValue());
     }
-    throw new IllegalStateException("Invalid pillar: " + transferExchange.getPillar());
+    throw new IllegalStateException("Invalid pillar: " + exchange.getPillar());
   }
 
-  private FundBalance fundBalance(
-      TransferExchange transferExchange, List<FundBalance> fundBalances) {
+  private FundBalance fundBalance(Exchange exchange, List<FundBalance> fundBalances) {
     return fundBalances.stream()
-        .filter(
-            fundBalance -> transferExchange.getSourceFund().getIsin().equals(fundBalance.getIsin()))
+        .filter(fundBalance -> exchange.getSourceFund().getIsin().equals(fundBalance.getIsin()))
         .findFirst()
         .orElse(FundBalance.builder().build());
   }
 
-  private List<TransferExchange> getPendingTransfers(Person person) {
-    return transferExchangeService.get(person).stream()
-        .filter(transferExchange -> transferExchange.getStatus().equals(PENDING))
+  private List<TransferApplication> getPendingTransferApplications(Person person) {
+    List<Application> applications = applicationService.getApplications(PENDING, person);
+    return applications.stream()
+        .filter(application -> application.getType() == TRANSFER)
```

Comment:
Can be a method on Application so the usage would be `filter(Application::isTransfer)`